$OpenBSD$

cxtFinished returns Maybe VerifyData in tls >= 2.0.0.
Wrap in MIN_VERSION_tls(2,0,0) CPP conditional.

--- src/Simplex/Messaging/Transport.hs
+++ src/Simplex/Messaging/Transport.hs
@@ -1,7 +1,9 @@
 {-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE PackageImports #-}
 {-# LANGUAGE DuplicateRecordFields #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE FlexibleInstances #-}
@@ -132,6 +134,7 @@
 import GHC.IO.Handle.Internals (ioe_EOF)
 import Network.Socket
 import qualified Network.TLS as T
+import Unsafe.Coerce (unsafeCoerce)
 import qualified Network.TLS.Extra as TE
 import qualified Paths_simplexmq as SMQ
 import qualified Simplex.Messaging.Crypto as C
@@ -373,7 +376,7 @@
 
 withTlsUnique :: forall c p. TransportPeerI p => T.Context -> (ByteString -> IO (c p)) -> IO (c p)
 withTlsUnique cxt f =
-  cxtFinished cxt
+  (fmap unsafeCoerce <$> cxtFinished cxt)
     >>= maybe (closeTLS cxt >> ioe_EOF) f
   where
     cxtFinished = case sTransportPeer @p of
